src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java

```diff
@@ -239,6 +243,68 @@ private static void createBranches(LfNetwork lfNetwork, LoadingContext loadingCo
+        connectedSets.forEach(LfNetworkLoaderImpl::fixDiscreteVoltageControls);
+    }
+
+    public static Graph<LfBus, LfBranch> getZeroImpedanceSubGraph(LfNetwork network) {
+        List<LfBranch> zeroImpedanceBranches = network.getBranches().stream()
+            .filter(LfNetworkLoaderImpl::isZeroImpedanceBranch)
+            .filter(b -> b.getBus1() != null && b.getBus2() != null)
+            .collect(Collectors.toList());
+
+        Graph<LfBus, LfBranch> zeroImpedanceSubGraph = new Pseudograph<>(LfBranch.class);
+        for (LfBranch branch : zeroImpedanceBranches) {
+            zeroImpedanceSubGraph.addVertex(branch.getBus1());
+            zeroImpedanceSubGraph.addVertex(branch.getBus2());
+            zeroImpedanceSubGraph.addEdge(branch.getBus1(), branch.getBus2(), branch);
+        }
+
+        return  zeroImpedanceSubGraph;
+    }
+
+    public static boolean isZeroImpedanceBranch(LfBranch branch) {
+        PiModel piModel = branch.getPiModel();
+        return piModel.getZ() < DcEquationSystem.LOW_IMPEDANCE_THRESHOLD;
+    }
+
+    private static void fixDiscreteVoltageControls(Set<LfBus> zeroImpedanceConnectedSet) {
+        // Get the list of discrete controlled buses in the zero impedance connected set
+        List<LfBus> discreteControlledBuses = zeroImpedanceConnectedSet.stream().filter(LfBus::isDiscreteVoltageControlled).collect(Collectors.toList());
+        // First resolve problem of mixed shared controls, that is if there are any generator/svc voltage control together with discrete voltage control(s)
+        // Check if there is one bus with remote voltage control or local voltage control
+        boolean hasControlledBus = zeroImpedanceConnectedSet.stream().anyMatch(lfBus -> !lfBus.getControllerBuses().isEmpty()
+                || (lfBus.hasVoltageControl() && lfBus.getControllerBuses().isEmpty()));
+        if (hasControlledBus) {
+            // If any generator/svc voltage controls, remove all discrete voltage controls
+            // TODO: deal with mixed shared controls instead of removing all discrete voltage controls
+            LOGGER.warn("Zero impedance connected set with voltage control and discrete voltage controls: only generator control is kept");
+            // First remove them from controllers
+            discreteControlledBuses.stream().map(LfBus::getDiscreteVoltageControl).flatMap(dvc -> dvc.getControllers().stream())
+                .forEach(controller -> controller.setDiscreteVoltageControl(null));
+            // Then remove them from controlled
+            discreteControlledBuses.forEach(lfBus -> lfBus.setDiscreteVoltageControl(null));
+        } else {
+            // Then resolve problem of discrete voltage controls
+            if (discreteControlledBuses.size() > 1) {
+                // We have several discrete controls whose controlled bus are in the same non-impedant connected set
```

Comment:
Because even if the target is the same, it leads to a KLU issue. The V of the second control bus is already fixed by a zero impedance branch equation, you cannot fix it twice with a discrete controlled bus.
.